src/mento.ts

```diff
@@ -262,47 +263,57 @@ export class Mento {
     return this.signerOrProvider.populateTransaction(tx)
   }
 
+  /** Returns the Broker contract */
+  getBroker(): IBroker {
+    return this.broker
+  }
+
   /**
    * Returns the list of exchanges available in Mento (cached)
    * @returns the list of exchanges
    */
-  private async getExchanges(): Promise<Exchange[]> {
+  async getExchanges(): Promise<Exchange[]> {
     if (this.exchanges.length > 0) {
       return this.exchanges
     }
 
-    const exchanges: Exchange[] = []
-
     const exchangeProvidersAddresses = await this.broker.getExchangeProviders()
-    for (const exchangeProviderAddr of exchangeProvidersAddresses) {
-      const exchangeManager: IExchangeProvider =
-        IExchangeProvider__factory.connect(
-          exchangeProviderAddr,
-          this.signerOrProvider
-        )
-      const exchangesInManager = await exchangeManager.getExchanges()
-      for (const exchange of exchangesInManager) {
-        assert(exchange.assets.length === 2, 'Exchange must have 2 assets')
-
-        exchanges.push({
-          providerAddr: exchangeProviderAddr,
-          id: exchange.exchangeId,
-          assets: exchange.assets,
-        })
-      }
-    }
+
+    const exchanges: Exchange[] = (
+      await Promise.all(
+        exchangeProvidersAddresses.map((a) => this.getExchangesForProvider(a))
+      )
+    ).flat()
 
     this.exchanges = exchanges
     return exchanges
   }
 
+  /**
+   * Returns the list of exchanges for a given provider address
+   * @returns list of exchanges
+   */
+  async getExchangesForProvider(providerAddr: Address): Promise<Exchange[]> {
```

Comment:
maybe private?

Reply:
IMHO there's no reason for any methods other than the constructor to be private. The interface of this class is already so simple that we're not doing users any favors by putting up artificial restrictions.